addons/addon-key-pair-mgmt-api/packages/key-pair-mgmt-services/lib/key-pair/schema/update-key-pair.js

```diff
@@ -12,6 +12,8 @@
  *  express or implied. See the License for the specific language governing
  *  permissions and limitations under the License.
  */
+const { nameRegex } = require('@amzn/base-services/lib/helpers/constants');
+
 const schema = {
   $schema: 'http://json-schema.org/draft-07/schema#',
   type: 'object',
```

Comment:
I am fairly certain `id` pattern should have a trailing `$`...which would be my b if I missed it when I updated this file. Obviously, the id wouldn't work to get anything if the id wasn't valid, but to be safe, let's make this `idRegex`. And add test

Reply:
test added and regex fixed